src/repositories/core.py

```diff
@@ -1,14 +1,85 @@
 from sqlalchemy.ext.asyncio import AsyncSession, async_sessionmaker, create_async_engine
+from typing import Any, Optional, Type
+from sqlalchemy import select, update, asc, desc
+from pydantic import BaseModel
 
 from src.core.config import settings
-from src.models.core import BaseModel
+from src.models.core import BaseModel as CoreBaseModel
+from src.ports.database_interface import DatabaseInterface
 
 
 class DBSessionMixin:
     def __init__(self, db: AsyncSession):
         self.db = db
 
 
+class BaseRepository(DatabaseInterface, DBSessionMixin):
+    model: Type[CoreBaseModel]
+
+    async def get_by_id(self, entity_id: Any):
+        stmt = select(self.model).where(self.model.id == entity_id)
+        result = await self.db.execute(stmt)
+        instance = result.scalars().first()
+
+        return instance
+
+    async def get_all(
+        self,
+        page: int = 1,
+        limit: int = 10,
+        order_by: str = "asc",
+        sort_by: Optional[str] = None,
+    ):
+        stmt = select(self.model)
+
+        if sort_by:
+            sort_column = getattr(self.model, sort_by, None)
+            if sort_column:
+                if order_by == "asc":
+                    stmt = stmt.order_by(asc(sort_column))
+                elif order_by == "desc":
+                    stmt = stmt.order_by(desc(sort_column))
```

Comment:
Line 37, redundant nesting. You can move `if sort_column` to same indent as `if sort_by`.

Reply:
For order by you can use match/case